homeassistant/components/dsmr/sensor.py

```diff
@@ -209,13 +230,17 @@ async def connect_and_reconnect():
 class DSMREntity(Entity):
     """Entity reading values from DSMR telegram."""
 
-    def __init__(self, name, obis, config):
+    def __init__(self, name, device_name, device_serial, obis, config):
         """Initialize entity."""
         self._name = name
         self._obis = obis
         self._config = config
         self.telegram = {}
 
+        self._device_name = device_name
+        self._device_serial = device_serial
+        self._unique_id = f"{device_serial}_{name}"
```

Comment:
```suggestion
        self._unique_id = f"{device_serial}_{name}".replace(' ', '_')
```

Reply:
there isnt really a required format but this males the string more like an id imo